Include/internal/pycore_gc.h

```diff
@@ -238,9 +240,11 @@ extern PyObject *_PyGC_GetObjects(PyInterpreterState *interp, Py_ssize_t generat
 extern PyObject *_PyGC_GetReferrers(PyInterpreterState *interp, PyObject *objs);
 
 // Functions to clear types free lists
+extern void _PyGC_ClearFreeList(PyInterpreterState *interp);
+extern void _Py_ClearFreeLists(_PyFreeListState *state);
```

Comment:
Hummm, I find this asymmetry a bit odd. Wouldn't be better to keep all functions receiving `PyInterpreterState *interp` and then fetching the freelist state from there instead?

That way the code of the caller will be also cleaner and doesn't need to deal with where the freelist is stored.

Reply:
In the free-threaded builds, we want to clear the free-lists for a specific thread (not all threads). Passing `interp` doesn't provide enough information. In some cases the thread we are clearing is not even the active thread, such as when we call `PyThreadState_Clear()` on dead threads after fork.